docs/rfcs/2024-08-06-json-datatype.md

```diff
@@ -0,0 +1,141 @@
+---
+Feature Name: Json Datatype
+Tracking Issue: https://github.com/GreptimeTeam/greptimedb/issues/4230
+Date: 2024-8-6
+Author: "Yuhan Wang <profsyb@gmail.com>"
+---
+
+# Summary
+This RFC proposes a method for storing and querying JSON data in the database.
+
+# Motivation
+JSON is widely used across various scenarios. Direct support for writing and querying JSON can significantly enhance the database's flexibility.
+
+# Details
+
+## User Interface
+The feature introduces a new data type for the database, similar to the common JSON type. Data is written as JSON strings and can be queried using functions.
+
+For example:
+```SQL
+CREATE TABLE IF NOT EXISTS test (
+    ts TIMESTAMP TIME INDEX,
+    a INT,
+    b JSON
+);
+
+INSERT INTO test VALUES(
+    0,
+    0,
+    '{
+        "name": "jHl2oDDnPc1i2OzlP5Y",
+        "timestamp": "2024-07-25T04:33:11.369386Z",
+        "attributes": { "event_attributes": 48.28667 }
+    }'
+);
+
+SELECT json_get(b, 'name') FROM test;
++---------------------+
+| b.name              |
++---------------------+
+| jHl2oDDnPc1i2OzlP5Y |
++---------------------+
+
+SELECT json_get_by_paths(b, 'attributes', 'event_attributes') + 1 FROM test;
```

Comment:
This looks like a downside to the current proposed implementation. We have to access fields by function. But if we leverage a more "native" type like `Struct` it would be much easier to use like 

```sql
select b.attributes.event_attributes from test
```

And `Struct` is more friendly to parquet

Reply:
I think we can provide the syntactic sugar for `JSON`?

e.g., 
```sql
select (b?.attributes?.event_attributes || default_value) from test
```

Can be treats as `get_path_or_default("b?.attributes?.event_attributes", default_value)`